crates/router/src/connector/payme.rs

```diff
@@ -428,87 +541,148 @@ impl ConnectorIntegration<api::Execute, types::RefundsData, types::RefundsRespon
+        Ok(format!(
+            "{}{}{}",
+            String::from_utf8_lossy(secret),
+            resource.payme_transaction_id,
+            resource.payme_sale_id
+        )
+        .as_bytes()
+        .to_vec())
+    }
+
+    fn get_webhook_object_reference_id(
+        &self,
+        request: &api::IncomingWebhookRequestDetails<'_>,
     ) -> CustomResult<api::webhooks::ObjectReferenceId, errors::ConnectorError> {
-        Err(errors::ConnectorError::WebhooksNotImplemented).into_report()
+        let resource: payme::WebhookEventDataResource =
+            request
+                .body
+                .parse_struct("WebhookEvent")
+                .change_context(errors::ConnectorError::WebhookBodyDecodingFailed)?;
+        let id = match resource.notify_type {
+            transformers::NotifyType::SaleComplete
+            | transformers::NotifyType::SaleAuthorized
+            | transformers::NotifyType::SaleFailure => api::webhooks::ObjectReferenceId::PaymentId(
+                api_models::payments::PaymentIdType::ConnectorTransactionId(resource.payme_sale_id),
+            ),
+            transformers::NotifyType::Refund => api::webhooks::ObjectReferenceId::RefundId(
+                api_models::webhooks::RefundIdType::ConnectorRefundId(resource.payme_sale_id),
+            ),
+            transformers::NotifyType::SaleChargeback
+            | transformers::NotifyType::SaleChargebackRefund => {
+                api::webhooks::ObjectReferenceId::PaymentId(
+                    api_models::payments::PaymentIdType::ConnectorTransactionId(
+                        resource.payme_sale_id,
+                    ),
+                )
+            }
+        };
+        Ok(id)
     }
 
     fn get_webhook_event_type(
         &self,
-        _request: &api::IncomingWebhookRequestDetails<'_>,
+        request: &api::IncomingWebhookRequestDetails<'_>,
     ) -> CustomResult<api::IncomingWebhookEvent, errors::ConnectorError> {
-        Err(errors::ConnectorError::WebhooksNotImplemented).into_report()
+        let resource: payme::WebhookEventDataResourceEvent = request
+            .body
+            .parse_struct("WebhookEvent")
+            .change_context(errors::ConnectorError::WebhookBodyDecodingFailed)?;
+        let event_type = match resource.notify_type {
+            transformers::NotifyType::SaleComplete => {
+                api::IncomingWebhookEvent::PaymentIntentSuccess
+            }
+            transformers::NotifyType::SaleFailure => {
+                api::IncomingWebhookEvent::PaymentIntentFailure
+            }
+            transformers::NotifyType::Refund => api::IncomingWebhookEvent::RefundSuccess,
+            transformers::NotifyType::SaleAuthorized
+            | transformers::NotifyType::SaleChargeback
+            | transformers::NotifyType::SaleChargebackRefund => {
+                api::IncomingWebhookEvent::EventNotSupported
+            }
+        };
```

Comment:
Can we move this to a `From` implementation instead, and move it to `transformers.rs`?